### PR TITLE
Temporarily revert #9439 multi-currency v2

### PR DESCRIPTION
**Pull request by ricardo**
Context: p1726755214216999/1726688493.292339-slack-C01BZUL57SQ

**TL;DR:** We didn't find any regressions other than #9455, but we would like to postpone the release of it.

Fixes 233-gh-Automattic/woopayments

#### Changes proposed in this Pull Request

Reverts #9439 / af3cc60d6e5dc41c20e384ce49e1cd05d7db89fa.

#### Testing instructions

- Smoke test multi-currency.
- Ensure #9386 are still applied to the old code.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `9468` or branch name `revert/9439-multi-currency-v2` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=revert/9439-multi-currency-v2) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 4ac9a98f8fd62f57acb64d0d8e6bad925a0e553b
- Build time: 2024-09-19 18:19:43 UTC

_Note: the build is updated when a new commit is pushed to this PR._

**Review by rafaelzaleski — APPROVED** [review]
Looks good and tests well!

✅ Smoke test multi-currency.
✅ Ensure Disable save changes button unless settings have changed  #9386 are still applied to the old code.
